rskj-core/src/main/java/co/rsk/cli/tools/DbMigrate.java

```diff
@@ -183,12 +187,39 @@ private void migrate(DbMigrationInformation dbMigrationInformation) {
 
         logger.info("Migrating data source: {}", sourceKeyValueDataSource.getName());
 
-        sourceKeyValueDataSource.keys().forEach(sourceKey ->
+        Object iterator = sourceKeyValueDataSource.iterator();
```

Comment:
What do you think about hide the specific Iterator implementations? 
We could have a common interface and specific wrapper for each database kind. 
With that we can avoid the cast of the different types and avoid the logic duplication as the content of both if else is very similar. 
Also the changes into specific iterators won't affect to this class.

Reply:
Done